ui.apps/src/main/content/jcr_root/apps/core/fd/components/form/fileinput/v1/fileinput/README.md

```diff
@@ -0,0 +1,83 @@
+
+### Edit Dialog Properties
+The following properties are written to JCR for this Form File component and are expected to be available as `Resource` properties:
+
+1. `./jcr:title` - defines the label to use for this field
+2. `./hideTitle` - if set to `true`, the label of this field will be hidden
+3. `./name` - defines the name of the field, which will be submitted with the form data
+4. `./default` - defines the default value of the field
+5. `./description` - defines a help message that can be rendered in the field as a hint for the user
+6. `./required` - if set to `true`, this field will be marked as required, not allowing the form to be submitted until the field has a value
+7. `./requiredMessage` - defines the message displayed as tooltip when submitting the form if the value is left empty
+8. `./readOnly` - if set to `true`, the filed will be read only
+9. `./multiSelection` - if set to `true`, the filed will allow to add multiple files in single selection or multiple selections
+10. `./minItems` - if value is selected/provided this will check for minimum number of files that can be attached
+11. `./maxItems` - if value is selected/provided this will check for maximum number of files that can be attached
+12. `./maxFileSize` - if value is selected/provided this will check for maximum file size allowed
+13. `./accept` - defines the type of files accepted to upload
+14. `./showComment` - if set to `true`, comments can be added to attachment
+15. `./minItemsMessage` - defines the message displayed as tooltip when submitting the form if less than allowed minimum files uploaded
+16. `./maxItemsMessage` - defines the message displayed as tooltip when submitting the form if more than allowed maximum files uploaded
+17. `./maxFileSizeMessage` - defines the message displayed as tooltip when submitting the form if the uploaded file size is greater than allowed
+18. `./acceptMessage` - defines the message displayed as tooltip when submitting the form if the uploaded file type is not allowed
+
+## Client Libraries
+The component provides a `core.forms.components.fileinput.v1` client library category that contains a JavaScript
+component. It should be added to a relevant site client library using the `embed` property.
+
+It also provides a `core.forms.components.fileinput.v1.editor` editor client library category that includes
+JavaScript handling for dialog interaction. It is already included by its edit dialog.
+
+## BEM Description
+```
+BLOCK cmp-adaptiveform-fileinput
+    ELEMENT cmp-adaptiveform-fileinput__label
+    ELEMENT cmp-adaptiveform-fileinput__widget
+    ELEMENT cmp-adaptiveform-fileinput__questionmark
+    ELEMENT cmp-adaptiveform-fileinput__shortdescription
+    ELEMENT cmp-adaptiveform-fileinput__longdescription
+    ELEMENT cmp-adaptiveform-fileinput__errormessage
+    ELEMENT cmp-adaptiveform-fileinput__minmessage
+    ELEMENT cmp-adaptiveform-fileinput__maxmessage
```

Comment:
we don't need to add the edit dialog classes here. only those classes should be added those can be styled in the actual rendition.